modules/zlib-ng/2.1.6/patches/zlib-ng-bazel-overlay.patch

```diff
@@ -0,0 +1,1626 @@
++            "arch/x86/slide_hash_avx2.c",
++            "arch/x86/slide_hash_sse2.c",
++            "arch/x86/x86_features.c",
++            "arch/x86/x86_features.h",
++            "arch/x86/x86_intrins.h",
++        ],
++        "//conditions:default": [],
++    }),
++    hdrs = select({
++        ":zlib-compat_enabled": [
++            ":zconf_h_gen",
++            ":zlib_h_gen",
++            ":zlib_name_mangling_h_gen",
++        ],
++        "//conditions:default": [
++            ":zconf-ng_h_gen",
++            ":zlib-ng_h_gen",
++            ":zlib_name_mangling-ng_h_gen",
++        ],
++    }),
++    copts = [
++        "-std=c11",
++
++        # Standard feature configuration.
++        "-DWITH_GZFILEOP",
++        "-DWITH_OPTIM",
++        "-DWITH_NEW_STRATEGIES",
++    ] + select({
++        ":zlib-compat_enabled": ["-DZLIB_COMPAT"],
++        "//conditions:default": [],
++    }) + select({
++        "@platforms//cpu:x86_64": [
++            "-D_LARGEFILE64_SOURCE=1",
++            "-D__USE_LARGEFILE64",
++            "-DX86_FEATURES",
++            "-DX86_AVX2",
++            "-mavx2",
++            "-DX86_AVX512",
++            "-mavx512f",
++            "-mavx512dq",
++            "-mavx512bw",
++            "-mavx512vl",
++            "-DHAVE_CASCADE_LAKE", # Applies to AMD and Intel CPUs.
++            "-mtune=cascadelake",
++            "-DX86_MASK_INTRIN",
++            "-DX86_AVX512VNNI",
++            "-mavx512vnni",
++            "-DX86_SSE42",
++            "-msse4.2",
++            "-DX86_SSE2",
++            "-msse2",
++            "-DX86_SSSE3",
++            "-mssse3",
++            "-DX86_PCLMULQDQ_CRC",
++            "-mpclmul",
++            "-DX86_VPCLMULQDQ_CRC",
++            "-mvpclmulqdq",
++            "-DHAVE_CASCADE_LAKE",
++        ],
++        "//conditions:default": [],
++    }) + select({
++        "@platforms//os:linux": [
++            "-DHAVE_ALIGNED_ALLOC",
++            "-DHAVE_SYS_AUXV_H",
++            "-DHAVE_UNISTD_H",
++            "-DHAVE_POSIX_MEMALIGN",
++        ],
++        "//conditions:default": [],
++    }) + [
++        # We can expect any non-ancient compiler to support these.
++        "-DHAVE_ATTRIBUTE_VISIBILITY_HIDDEN",
++        "-DHAVE_ATTRIBUTE_VISIBILITY_INTERNAL",
++        "-DHAVE_ATTRIBUTE_ALIGNED",
++        "-DHAVE_BUILTIN_CTZ",
++        "-DHAVE_BUILTIN_CTZLL",
++    ],
++    visibility = ["//visibility:public"],
++)
+diff --git a/MODULE.bazel b/MODULE.bazel
+new file mode 100644
+index 00000000..b791f476
+--- /dev/null
++++ b/MODULE.bazel
+@@ -0,0 +1,9 @@
++module(
++    name = "zlib-ng",
++    version = "2.1.6",
++    compatibility_level = 1,
++)
++
++bazel_dep(name = "bazel_skylib", version = "1.5.0")
++bazel_dep(name = "platforms", version = "0.0.8")
++bazel_dep(name = "rules_cc", version = "0.0.9")
+diff --git a/MODULE.bazel.lock b/MODULE.bazel.lock
+new file mode 100644
+index 00000000..8f382606
+--- /dev/null
++++ b/MODULE.bazel.lock
```

Comment:
you shouldn't need to patch in the lockfile here (it doesn't do anything anyway). similarly for the github workflow yaml files. Could you maybe add a file filter to whatever script you generated this patch with?